bundle/deploy/terraform/unbind.go

```diff
@@ -25,18 +26,44 @@ func (m *unbind) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 		return diag.Errorf("terraform init: %v", err)
 	}
 
-	err = tf.StateRm(ctx, fmt.Sprintf("%s.%s", m.resourceType, m.resourceKey))
+	err = tf.StateRm(ctx, fmt.Sprintf("%s.%s", m.tfResourceType, m.resourceKey))
 	if err != nil {
 		return diag.Errorf("terraform state rm: %v", err)
 	}
 
+	// Also remove the permission if it exists
+	// First check terraform state list to see if the permission exists
+	state, err := tf.Show(ctx)
+	if err != nil {
+		return diag.Errorf("terraform show: %v", err)
+	}
+
+	if state.Values == nil || state.Values.RootModule == nil || state.Values.RootModule.Resources == nil {
+		return nil
+	}
+
+	for _, resource := range state.Values.RootModule.Resources {
+		if resource.Address == fmt.Sprintf("databricks_permissions.%s_%s", m.bundleType, m.resourceKey) {
+			err = tf.StateRm(ctx, resource.Address)
+			if err != nil {
+				return diag.Errorf("terraform state rm: %v", err)
+			}
+		}
+		if resource.Address == fmt.Sprintf("databricks_grants.%s_%s", m.bundleType, m.resourceKey) {
+			err = tf.StateRm(ctx, resource.Address)
+			if err != nil {
+				return diag.Errorf("terraform state rm: %v", err)
+			}
```

Comment:
Removal is identical for both, you could test the resource address for both paths in a single condition.